addons/google_calendar/models/res_users.py

```diff
@@ -36,12 +36,12 @@ def _google_calendar_authenticated(self):
 
     def _get_google_calendar_token(self):
         self.ensure_one()
-        if self._is_google_calendar_valid():
+        if self.google_calendar_rtoken and not self._is_google_calendar_valid():
             self._refresh_google_calendar_token()
         return self.google_calendar_token
 
     def _is_google_calendar_valid(self):
-        return self.google_calendar_token_validity and self.google_calendar_token_validity < (fields.Datetime.now() + timedelta(minutes=1))
+        return self.google_calendar_token_validity and self.google_calendar_token_validity >= (fields.Datetime.now() + timedelta(minutes=1))
```

Comment:
Funny one :D was it introduced in 14.1 only ?

Reply:
So actually it's more or less working without my fix, 
_is_google_calendar_valid was returning true when you had a token validity (and thus most likely a rtoken ) and the validity was outdated. so the correct name for that method should have been is_calendar_rtoken_valid_but_not_the_calendar_token.

So it's actually mostly a semantic fix. 

But I agree some change should probably be backported in previous version

Reply:
Ok it is called only there. Grepping code it is used only once, why not integrating it directly in code to lessen call chain ?

Reply:
for the same reason we use intermediate variable: to give meaning to some piece of code in order to ease the reading. But the question is indeed interesting to ask. 

and to be really bad liver, the 14.1 is a stable version I don't want to remove a method that may be use in some customization.